lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -207,37 +204,38 @@ spec = do
         verify r
             [ expectSuccess
             , expectResponseCode HTTP.status202
-            , expectFieldBetween
-                    (#amount . #getQuantity) (feeMin + amt, feeMax + amt)
-            , expectFieldEqual (#direction . #getApiT) Outgoing
-            , expectFieldEqual (#status . #getApiT) Pending
+            , expectFieldSatisfy (#amount . #getQuantity) (>= feeMin + amt)
+            , expectFieldSatisfy (#amount . #getQuantity) (<= feeMax + amt)
```

Comment:
The direction here is very nice. But unfortunately the failure-descriptions are now unhelpful:
```
Failures:

  src/Test/Integration/Scenario/API/Transactions.hs:195:5:
  1) API Specifications TRANS_CREATE_01 - Single Output Transaction
       uncaught exception: IOException of type UserError
       user error (predicate failed for: 24)

  To rerun use: --match "/API Specifications/TRANS_CREATE_01 - Single Output Transaction/"
```

I think we need to capture the same `between` we had before while keeping it as composable as you have now, by making the predicate return failure reasons. (Not just `a -> Bool`)

At the call-site, maybe it could look like this:

```haskell
expectFieldSatisfy 
    (#amount . #getQuantity) 
    (between (feeMin + amt, feeMax + amt))
```
or some variant as you prefer

Reply:
That is a fair point, for the current `expectFieldBetween` the failulre message is:

```
  src/Test/Integration/Scenario/API/Transactions.hs:198:5: 
  1) API Specifications TRANS_CREATE_01x - Single Output Transaction
       uncaught exception: IOException of type UserError
       user error (expected 24 <= 23)

```

So we are missing the expected condition now.

I think the idea of #1238 was, for now, to replace exisitng expactations with `expect...Satisfy` to make DSL lighter.
On top of those changes we may start introducing improvements. (in subsequent PRs)

Good point also on makeing the methods shorter. I have change that in -> 2b2c51dc3a4775d88033c6e97d2c4c73f90bf332
I also like the idea of `between` and having:

```
expectField #amount (between (feeMin, feeMax))
``` 
Which, indeed, looks neat. 

Again, I would postpone it to the subsequent PRs. Thoughts?

Reply:
> Again, I would postpone it to the subsequent PRs. Thoughts?

My concern would be that failing tests will be more difficult to debug before we fix that.

I also suspect it would be easier to go straight from `expectFieldBetween` to `expectField #foo (between (a, b))`, than converting to two `expectFieldSatisfy` checks (`>=`, and `<=`)

Reply:
I think we would only need:
- `equal` - we can re-use `Test.Hspec.Expectations.shouldBe`
- `notEqual` - we can re-use `Test.Hspec.Expectations.shouldNotBe`
- `between :: (a, a) -> a -> IO ()` the only one we need to implement ourselves